client/middleware/restricted.ts

```diff
@@ -0,0 +1,8 @@
+export default defineNuxtRouteMiddleware((to, from) => {
+  const { user } = storeToRefs(useAuth())
+
+  if (to.path.startsWith('/configuration') && user.value?.role === 'REGULAR') {
+    if (!from.path.startsWith('/configuration')) return abortNavigation()
```

Comment:
I don't think it's a good idea to depend on the configuration page here, especially when you use that middleware explicitly only in the configuration.vue file.

What I mean is that if there are other admin restricted pages, we should just need to add the middleware to the list for that page, not change this code.